server/services/conditionService.ts

```diff
@@ -86,42 +92,30 @@ export default class ConditionService {
     licenceType: string,
     version: string = null
   ): Promise<{ category: string; conditions: AdditionalConditionAp[] | AdditionalConditionPss[] }[]> {
-    let policyConditions: LicencePolicy
-    if (version) {
-      policyConditions = await this.getConditions(version)
-    } else {
-      policyConditions = await this.getActiveConditions()
-    }
+    const additionalConditions = await this.getAdditionalConditions(version)
     const map = new Map()
-    policyConditions.additionalConditions[licenceType].forEach(
-      (condition: AdditionalConditionAp | AdditionalConditionPss) => {
-        const collection = map.get(condition.category)
-        if (!collection) {
-          map.set(condition.category, [condition])
-        } else {
-          collection.push(condition)
-        }
+    additionalConditions[licenceType].forEach((condition: AdditionalConditionAp | AdditionalConditionPss) => {
+      const collection = map.get(condition.category)
+      if (!collection) {
+        map.set(condition.category, [condition])
+      } else {
+        collection.push(condition)
       }
-    )
+    })
```

Comment:
meh.. if you're looping over something to build up an object, a `reduce` would be a bit more idiomatic.

Reply:
This isn't actually anything I've changed in this PR, just a formatting change making GH think it's new code. I can certainly look into changing it to a reduce though!

Reply:
ah cool, no worries I misread it